src/shared/get-agent-for-url.js

```diff
@@ -46,6 +48,9 @@ export const getAgentForURL = (url: URL): HttpAgent | HttpsAgent => {
             return new https.Agent(agentOptions);
 
         default:
-            throw new Error(`Unsupported protocol: ${url.protocol}`);
+            throw new KAError(
+                `Unsupported protocol: ${url.protocol}`,
+                Errors.Internal,
```

Comment:
Maybe InvalidInput?

Reply:
Possibly. I could see how this may change based on input to the gateway itself. Seems reasonable.